store/src/java/com/zimbra/cs/lmtpserver/ZimbraLmtpBackend.java

```diff
@@ -339,11 +347,21 @@ public void deliver(LmtpEnvelope env, InputStream in, int sizeHint) throws Unrec
         Blob blobEEW = null;
         // duplicating input stream for EEW
         try {
-            final String content = IOUtils.toString(in); // pop content from original input stream
-            IOUtils.closeQuietly(in);
-            in = IOUtils.toInputStream(content); // push content back to input stream
-            final String contentEEW = ExternalEmailWarning.getInstance().getUpdatedContent(content);
-            inEEW = IOUtils.toInputStream(contentEEW);
+            for (int i = 0; i < CHARSET_NAMES.length; i++) {
+                final String charsetName = CHARSET_NAMES[i];
+                // copy input stream and evaluate content encoding
+                final byte[] ba = IOUtils.toByteArray(in);
+                IOUtils.closeQuietly(in);
+                in = new ByteArrayInputStream(ba);
+                final String content = new String(ba, charsetName);
+                // if current encoding is suitable, use it; otherwise, continue or use default
+                if (!content.contains(GARBLED_CHARACTER) || i == CHARSET_NAMES.length - 1) { }
```

Comment:
I doesn't mean this. You should write this only.
`if (!content.contains(GARBLED_CHARACTER) || i == CHARSET_NAMES.length - 1) { 
       final String contentEEW = ExternalEmailWarning.getInstance().getUpdatedContent(content);
       inEEW = IOUtils.toInputStream(contentEEW, charsetName);
       break;
}`

Reply:
Yes....expecting this without else bock

Reply:
@zimsuchitgupta @shrutig0510 , Could you please check now